packages/nuxt/src/app/composables/loading-indicator.ts

```diff
@@ -0,0 +1,106 @@
+import { computed, onBeforeUnmount, ref } from 'vue'
+import { useNuxtApp } from '#app/nuxt'
+
+export type LoadingIndicatorOpts = {
+  duration: number,
+  throttle: number
+}
+
+/**
+ * composable to handle the loading state of the page
+ */
+export function useLoadingIndicator (opts: Partial<{
+    duration: number,
+    throttle: number
+  }> = {}) {
+  const { duration = 2000, throttle = 200 } = opts
+  const nuxtApp = useNuxtApp()
+  const progress = ref(0)
+  const isLoading = ref(false)
+  const step = computed(() => 10000 / duration)
+
+  let _timer: any = null
+  let _throttle: any = null
+
+  function start () {
```

Comment:
do you think we could remove some of these functions out of `useLoadingIndicator` so they don't have be re-instantiated each time the composable is called?

Reply:
Sure !

Reply:
Sure, i'll try :+1: but all functions are closures. I'm not sure what it's going to be in terms of readability.